src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesComputer.java

```diff
@@ -40,6 +56,39 @@ public void taskCompletedWithProblems(Executor executor, Queue.Task task, long d
         LOGGER.log(Level.FINE, " Computer " + this + " taskCompletedWithProblems");
     }
 
+    @Exported
+    public List<Container> getContainers() throws UnrecoverableKeyException, CertificateEncodingException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        KubernetesSlave slave = getNode();
+        if(slave == null) {
+            return new ArrayList<>();
+        }
+
+        KubernetesCloud cloud = slave.getKubernetesCloud();
+        KubernetesClient client = cloud.connect();
+
+        String namespace = StringUtils.defaultIfBlank(slave.getNamespace(), client.getNamespace());
+        Pod pod = client.pods().inNamespace(namespace).withName(getName()).get();
+
+        return pod.getSpec().getContainers();
+    }
+
+    public void doContainerLog(@QueryParameter String containerId,
+                               StaplerRequest req, StaplerResponse rsp) throws UnrecoverableKeyException, CertificateEncodingException, NoSuchAlgorithmException, KeyStoreException, IOException {
+        ByteBuffer outputStream = new ByteBuffer();
```

Comment:
No permission check here, the log becomes accessible to any user who is able to read the agent

Reply:
this ?

    Jenkins.get().checkPermission(Jenkins.ADMINISTER);

Reply:
Already add permission check.